pkg/cli/userfiletable_test.go

```diff
@@ -98,6 +249,126 @@ func checkUserFileContent(
 	require.True(t, bytes.Equal(got, expectedContent))
 }
 
+func TestUserFileUploadRecursive(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	c := NewCLITest(TestCLIParams{T: t})
+	defer c.Cleanup()
+	c.omitArgs = true
+
+	tmpDir, cleanFn := testutils.TempDir(t)
+	defer cleanFn()
+	testDir := filepath.Join(tmpDir, "testdir")
+	require.NoError(t, os.MkdirAll(filepath.Join(testDir, "d1/d11/d111/d1111"), 0755))
+	require.NoError(t, os.MkdirAll(filepath.Join(testDir, "d2/d21"), 0755))
+
+	// Create a nontrivial subtree under 'testdir' by creating and writing the
+	// following five files with different contents:
+	// testdir/d1/test-temp-prefix-f1.csv
+	// testdir/d1/d11/test-temp-prefix-f2.csv
+	// testdir/d1/d11/test-temp-prefix-f3.csv
+	// testdir/d1/d11/d111/d1111/test-temp-prefix-f4.csv
+	// testdir/d2/d21/test-temp-prefix-f5.csv
+	files := []struct {
+		relativePath string
+		content      []byte
+	}{
+		{
+			"d1/" + testTempFilePrefix + "f1.csv",
+			[]byte("content111"),
+		},
+		{
+			"d1/d11/" + testTempFilePrefix + "f2.csv",
+			[]byte("content222"),
+		},
+		{
+			"d1/d11/" + testTempFilePrefix + "f3.csv",
+			[]byte("content333"),
+		},
+		{
+			"d1/d11/d111/d1111/" + testTempFilePrefix + "f4.csv",
+			[]byte("content444"),
+		},
+		{
+			"d2/d21/" + testTempFilePrefix + "f5.csv",
+			[]byte("content555"),
+		},
+	}
+	for _, f := range files {
+		require.NoError(t,
+			ioutil.WriteFile(filepath.Join(testDir, f.relativePath), f.content, 0666))
+	}
+
+	ctx := context.Background()
+
+	for _, tc := range []struct {
+		name                 string
+		destination          string
+		srcWithTrailingSlash bool
+	}{
+		{
```

Comment:
nit: instead of duplicating test cases can we restructure it to add a loop inside the outer loop that ranges over the testcases, and runs each test for `true` and `false`?

```
for srcWithTrailingSlash := range []bool{true, false} {
     t.Run(fmt.Sprintf("%s_withTrailingSlash=%t", tc.Name, srcWithTrailingSlash), func() { ...}}
```